components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt

```diff
@@ -298,26 +298,30 @@ class PromptFeature private constructor(
                     arrayOf(it?.content?.promptRequests, it?.content?.loading)
                 }
                 .collect { state ->
-                    state?.content?.let {
-                        if (it.promptRequests.lastOrNull() != activePromptRequest) {
+                    state?.content?.let { content ->
+                        if (content.promptRequests.lastOrNull() != activePromptRequest) {
                             // Dismiss any active select login or credit card prompt if it does
                             // not match the current prompt request for the session.
                             if (activePromptRequest is SelectLoginPrompt) {
                                 loginPicker?.dismissCurrentLoginSelect(activePromptRequest as SelectLoginPrompt)
+                            } else if (activePromptRequest is SaveLoginPrompt) {
+                                (activePrompt?.get() as? SaveLoginDialogFragment)?.dismissAllowingStateLoss()
                             } else if (activePromptRequest is SelectCreditCard) {
                                 creditCardPicker?.dismissSelectCreditCardRequest(
                                     activePromptRequest as SelectCreditCard
                                 )
                             }
 
                             onPromptRequested(state)
-                        } else if (!it.loading) {
+                        } else if (content.promptRequests.find { it is SaveLoginPrompt } == null) {
+                            (activePrompt?.get() as? SaveLoginDialogFragment)?.dismissAllowingStateLoss()
```

Comment:
I'm not sure if we want to keep this part. As we are observing both `promptRequests`  and `loading`, this will cause that when we reload or go to another site the `SaveLoginDialogFragment` will be dismissed. I think this is not the expected behavior at least in desktop the prompt is visible until the user interact with it or click on the web content.